### ISSUE TITLE
Help with updating TFT

### ISSUE THREAD

**Issue by StevenB1701** [Blackymas/NSPanel_HA_Blueprint#265]
Hi I've been able to flash esphome on my nspanel and started the tft update. However this took over 12 hours and when finished it just came up with an update failed message (System Data ERROR). Sensors from the NSP are showing in HA but all attempts to update the tft either via HA (service call) and esphome have so far failed. I have deleted blueprint and esphome (nspane) from HA and reflashed/reinstalled blueprint etc but all to no avail. Can you help? Thanks

**Comment by trunneml** [comment]
Hi, 

I had a similar problem.  I could fix the problem by downloading the `esphome_nspanel.yaml` and disable/comment  `captive_portal` and  `web_server` (lines 66-74):

```
##### TYPE OF ESP BOARD #####
esp32:
  board: esp32dev

# captive_portal:

# web_server:
#   port: 80
#   auth:
#     username: admin
#     password: ${wifi_password}
#     ##### advanced config - change to use web_password ##### 
#     # password: ${web_password}

##### OTA PASSWORD #####
ota:
  password: ${wifi_password}
```

After that I could upload the display firmware. It seems that the esp runs out of memory. I have the feeling the latest esphome Version needs more ram than the initial version where the upload was tested. Maybe some maintainer could test it or help us debugging it.

Maybe the problem is also related to  #204. Do you use HTTPS to provide the `tft` file? If so you could also try using HTTP (not HTTPS).

**Comment by danwright87** [comment]
While the web server does consume some of the ESPs limited memory, upload has been successful on latest version of ESPHome on my 9 panels and I believe q of the other Devs is on the latest aswell. 

The most common issues with upload are:
1) URL set to GitHub instead of local server
2) using HTTPs, use HTTP instead
3) Issues with your Wi-Fi

**Comment by deejaybeam** [comment]
runs well on my site too with latest version and dev versions. But just alone that it runs for 12h is strange if you read the wiki and use local http - this should run in ~10 mins.
Web-server and capacitive portal is a good alternitive to upload esp-firmware in many cases!

**Comment by trunneml** [comment]
I will try HTTP tomorrow. With HTTPS and API encryption the upload runs into some problems and the ESP reboots. 
The display still shows the upload screen and the percentage sometimes still increase. 😮

Maybe you should add a not that HTTPS makes problems. (My HA installation uses HTTPS directly on Port 8123)

**Comment by Blackymas** [comment]
Https is not necessarily the cause but a component of various issues.

I myself have 7 panels via https and via Github and all run without problems.

However, we have adapted the Esphome code in the wiki so that it points directly to HA with HTTP.

**Comment by StevenB1701** [comment]
Hi everyone and thanks for the suggestions. I have 3 NSPanels and decisded to try and flash the tft file from Nextion Editor. This worked on one and I started to set it up from the blueprint. But! After adding some more light controls it stuck on the initializing screen. Ive checked the panel name hasnt changed etc but it remains stuck on the initializing screen. The sceond nspanel would not communicate with the nextion editor so remains 'bricked'. With the third one I followed the instuction to the letter, double checking every stage. But it will not update the tft from esphome. (even via http). So frustration as I actually had one working for a short while :)

**Comment by sygad** [comment]
I've just come through this problem.

I set up a quick web server on my NAS to I could serve the file via HTTP. MY instance of HA uses HTTPS as I use DuckDNS and this I believe was the cause of most of my issues.

I also remvoed anything other than lowercase letters from my esp and blueprint name, probably a coincidence but it got me moving forward.

Finally, I had some odd issues with the sequence of events, I worked out in the end that I had to follow this sequence for it to work:

Download blueprint
Create ESP project (only lowercase letters)
Flash, Discover and Integrate into HA
Create automation from blueprint (ensure name is EXACTLY the same)
"Update TFT" from newly integrated device page

Coming from my local web server, it was VERY quick, <3mins.

I also came to the conclusion that if it takes more than an hour, it 100% failed, so no point in waiting 20hours to see if it will work, I am 99.99% confident, it won't.

**Comment by StevenB1701** [comment]
hi everyone. I've got all 3 of my NSPanels working now. I first flashed esphome to the panel. I then soldered a small wire to the TFT TX and RX pads and then using Nextion Editor I flashed the blank tft file, then flashed the nspanel.tft file. I then put them all into Home Assistant and setup the Blueprint. Using the "Update TFT display" toggle in esphome (on HA) the panels stated to do a small update and all works fine. Thanks to everyone for help
